src/client.ts

```diff
@@ -180,6 +192,73 @@ export default class Client {
     );
   }
 
+  private async setupCustomGemfile() {
+    // If we're working on the ruby-lsp itself, we can't create a custom Gemfile or we'd be trying to activate the same
+    // gem twice
+    if (this.workingFolder.endsWith("ruby-lsp")) {
+      return;
+    }
+
+    // Create the .ruby-lsp directory if it doesn't exist
+    const rubyLspDirectory = path.join(this.workingFolder, ".ruby-lsp");
+
+    if (!fs.existsSync(rubyLspDirectory)) {
+      fs.mkdirSync(rubyLspDirectory);
+    }
+
+    // Ignore the .ruby-lsp directory automatically
+    const gitIgnorePath = path.join(this.workingFolder, ".git/info/exclude");
+
+    if (!(await readFile(gitIgnorePath)).includes(".ruby-lsp")) {
+      fs.appendFileSync(gitIgnorePath, ".ruby-lsp");
+    }
+
+    // Generate the custom Gemfile that includes the `ruby-lsp`
+    const gemfilePath = path.join(rubyLspDirectory, "Gemfile");
+    const gemfile = [
+      "# This custom gemfile is automatically generated by the Ruby LSP extension.",
+      "# It should be automatically git ignored, but in any case: do not commit it to your repository.",
+      "",
+      'source "https://rubygems.org"',
+      "",
+    ];
+
+    // Only try to evaluate the top level Gemfile if there is one. Otherwise, we'll just create our own Gemfile
+    if (fs.existsSync(path.join(this.workingFolder, "Gemfile"))) {
+      gemfile.push('eval_gemfile "../Gemfile"');
+
+      // If the `ruby-lsp` exists in the bundle, add it to the custom Gemfile commented out
+      if (await this.migrateFromIncludingInBundle()) {
+        gemfile.push(
+          "# Uncomment the following line after following the instructions"
+        );
+        gemfile.push(
+          "# at https://github.com/Shopify/vscode-ruby-lsp#migrating-from-bundle"
+        );
+        gemfile.push('# gem "ruby-lsp", require: false, group: :development');
+      } else {
+        gemfile.push('gem "ruby-lsp", require: false, group: :development');
+      }
```

Comment:
Don't we need this code to execute outside of the check for the existence of the `Gemfile`? I guess if the project is not using a `Gemfile`, we still want to add `ruby-lsp` to the `Gemfile` that we are constructing.